### PR TITLE
Events

### PR DESCRIPTION
**Pull request by darmhoo**
Upload event gallery done

### CODE REVIEW

**Review by Golden-Ogbeka — CHANGES_REQUESTED** [review]
@darmhoo when uploading images, I noticed it's the file path of the image that is being saved not the cloudinary link.
When is the image uploaded to cloudinary.
I didn't see the parser function in use in the route

File: functions/cloudinary.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+dotenv.config();
+
+cloudinary.config({
+    cloud_name: process.env.CLOUD_NAME,
+    api_key: process.env.CLOUDINARY_API_KEY,
+    api_secret: process.env.CLOUDINARY_API_SECRET
+});
+
+const cloudStorage = new CloudinaryStorage({
```
Comment (darmhoo) [inline functions/cloudinary.ts:14]: This is the file responsible for the uploads. it serves as a middleware to check the validity of files sent and also uploads at the same time.

File: api/v1/routes/event.ts [hunk 2]
```diff
@@ -177,5 +179,18 @@ router.post(
 	],
 	EventController.RegisterEvent
 )
+router.post('/:id/upload', [parser.array('images'), header('x-api-key', 'API Access Denied')
```
Comment (darmhoo) [inline api/v1/routes/event.ts:182]: the parser is doing the job in the routes

File: api/v1/controllers/event.ts [hunk 3]
```diff
@@ -255,6 +256,30 @@ export default () => {
...
+    const UploadEventImages = async (req: express.Request<{ id: string }>,
+        res: express.Response) => {
+        try {
+            const { id } = req.params;
+            const event = await EventsModel.findById(id);
+            if (!event) return res.status(404).json({ message: 'Event not found' });
+            const fileArray: any = req.files;
+            if (!fileArray) return res.status(404).json({ message: "No file to upload" });
+            for (let i: number = 0; i < fileArray.length; i++) {
+                event.gallery.push(fileArray[i].path)
```
Comment (darmhoo) [inline api/v1/controllers/event.ts:268]: It is the cloudinary link. 
`"https://res.cloudinary.com/dizmz1ylh/image/upload/v1674304347/demo/ds91q1gdyzulbkobjflj.jpg",
            "https://res.cloudinary.com/dizmz1ylh/image/upload/v1674304347/demo/dwoazsj0fwzkgpsteptr.jpg",
            "https://res.cloudinary.com/dizmz1ylh/image/upload/v1674304347/demo/ubqrjy0fi8l9fciw9jxb.jpg"`

Comment (darmhoo) [reply api/v1/controllers/event.ts:268]: I uploaded those 3 with the api

Comment (Golden-Ogbeka) [reply api/v1/controllers/event.ts:268]: Ohh interesting. Glad it works

**Comment by Golden-Ogbeka** [comment]
I missed the Middleware. Maybe it's because parser is also in the controllers but I didn't see it used there

**Review by Golden-Ogbeka — APPROVED** [review]

**Pull request merged**